### PR TITLE
 disconnect and editing sessions limited, settings button moved

### PR DESCRIPTION
**Pull request by bergano65**

### CODE REVIEW

File: CogsMinimizer/Views/Home/Index.cshtml [hunk 1]
```diff
@@ -77,16 +77,19 @@
                                },
                                new { @class = "btn btn-warning btn-xs" })
 
-                                <small>&nbsp;&nbsp;</small>
+                                if (@subscription.ConnectedBy == User.Identity.Name)
```
Comment (shklar) [inline CogsMinimizer/Views/Home/Index.cshtml]: Blocking should be done not only in the UI (which is what you did), but also in the Controller, so that if someone just copy-pastes a URL and changes the subscription ID they will get an error message.

File: CogsMinimizer/Views/Subscription/Analyze.cshtml [hunk 2]
```diff
@@ -239,10 +239,12 @@
                                    DisplayName = Model.SubscriptionData.DisplayName,
                                    ServicePrincipalObjectId = ""
                                },
-                               new { @class = "btn btn-warning btn-xs" })
+                               new { @class = "btn btn-warning" })
```
Comment (shklar) [inline CogsMinimizer/Views/Subscription/Analyze.cshtml]: Please use a more "rounded" look for this button. Currently it looks like a simple rectangle.

File: CogsMinimizer/Views/Subscription/GetSettings.cshtml [hunk 3]
```diff
@@ -102,14 +102,24 @@ new SelectListItem
   </table>
 <table>
     <tr>
+        @if (Model.ConnectedBy == User.Identity.Name)
```
Comment (shklar) [inline CogsMinimizer/Views/Subscription/GetSettings.cshtml]: Same comment about validating in the controller that when someone clicks Save then we only accept it if it is the subscription owner

**Comment by bergano65** [comment]
Please look fixes (operations check in controller also, button rounded, tool tips added).
Next submit:

-          Exception handling, connect result check, at analyze check when subscription connected.

File: CogsMinimizer/Controllers/SubscriptionController.cs [hunk 4]
```diff
@@ -40,6 +40,12 @@ public ActionResult SaveSettings(string ServicePrincipalObjectId, Subscription s
                 Subscription existingSubscription =
                     dataAccess.Subscriptions.Where<Subscription>(s => s.Id.Equals(subscription.Id)).FirstOrDefault();
 
+                string currentUser = AzureAuthUtils.GetSignedInUserUniqueName();
+                if (currentUser != existingSubscription.ConnectedBy)
+                {
+                    throw new ArgumentException(string.Format("You can not edit settings for subscription that was connected by {0}", existingSubscription.ConnectedBy));
```
Comment (shklar) [inline CogsMinimizer/Controllers/SubscriptionController.cs]: Please rephrase:
"You are not authorized to edit the subscription settings. Please contact the subscription owner" 
Do not provide the name of the owner. It is available from the home page.

File: CogsMinimizer/Controllers/SubscriptionController.cs [hunk 5]
```diff
@@ -101,6 +107,12 @@ public ActionResult ReserveResource(string ResourceId, string SubscriptionId)
                 var resource = db.Resources.FirstOrDefault(x => x.SubscriptionId.Equals(SubscriptionId) && x.Id.Equals(ResourceId));
                 var subscription = db.Subscriptions.FirstOrDefault(x => x.Id.Equals(SubscriptionId));
 
+                string currentUser = AzureAuthUtils.GetSignedInUserUniqueName();
+                if (currentUser != subscription.ConnectedBy)
```
Comment (shklar) [inline CogsMinimizer/Controllers/SubscriptionController.cs]: This is wrong.
Anyone can reserve a resource or Extend a resource.
This is not limited to the subscription owner.

File: CogsMinimizer/Controllers/SubscriptionController.cs [hunk 6]
```diff
@@ -130,6 +142,12 @@ public ActionResult ExtendResource(string ResourceId, string SubscriptionId)
                 var subscription = db.Subscriptions.FirstOrDefault(x => x.Id.Equals(SubscriptionId));
 
                 // add subscription not found handling
+                string currentUser = AzureAuthUtils.GetSignedInUserUniqueName();
```
Comment (shklar) [inline CogsMinimizer/Controllers/SubscriptionController.cs]: Same comment. Anyone can Extend a resource.

File: CogsMinimizer/Controllers/SubscriptionController.cs [hunk 7]
```diff
@@ -255,13 +273,19 @@ public ActionResult ExtendExpired(string subscriptionId)
         {
             using (var db = new DataAccess())
             {
+                // add subscription not found handling
```
Comment (shklar) [inline CogsMinimizer/Controllers/SubscriptionController.cs]: Please mark the ExtendExpired action as private. We no longer expose it via the UI.
Same for Delete and DeleteExpired

File: CogsMinimizer/Controllers/SubscriptionController.cs [hunk 8]
```diff
@@ -306,13 +330,19 @@ public ActionResult Connect([Bind(Include = "Id, OrganizationId, DisplayName")]
...
             if (ModelState.IsValid)
-            {
+            {             
                 AzureResourceManagerUtil.RevokeAllRolesFromServicePrincipalOnSubscription(servicePrincipalObjectId, subscription.Id, subscription.OrganizationId);
                 
                 // add subscription not found handling
                 Subscription s = db.Subscriptions.Find(subscription.Id);
                 if (s != null)
                 {
+                    string currentUser = AzureAuthUtils.GetSignedInUserUniqueName();
```
Comment (shklar) [inline CogsMinimizer/Controllers/SubscriptionController.cs]: The validation that this is the subscription owner should happen earlier in the method, because by now we have revoked the permissions to the application.
RevokeAllRoles should happen only after the validation

File: CogsMinimizer/Views/Subscription/Analyze.cshtml [hunk 9]
```diff
@@ -321,8 +328,10 @@
                         @Html.DisplayFor(modelItem => item.Status)
                     </td>
                     <td>
-                        <button class="btn-info btn-xs" onclick="OnExtendResource(this, '@item.Id', '@item.SubscriptionId')">Extend</button>
-                        <button class="btn-success btn-xs" onclick="OnReserveResource(this, '@item.Id', '@item.SubscriptionId')">Reserve</button>
+                        <button class="btn-info btn-xs" onclick="OnExtendResource(this, '@item.Id', '@item.SubscriptionId')"
+                                data-toggle = "tooltip", title = "Prolong resource usage at number days specified in subscription settings.">Extend</button>
```
Comment (shklar) [inline CogsMinimizer/Views/Subscription/Analyze.cshtml]: Please rephrase:
Extend:
"Extend expiration date by a short period of time. See subscription settings" 
Reserve
"Extend expiration date by a long period of time. See subscription settings"

**Comment by bergano65** [comment]
Pls inspect changes submitted fixing your corrections, in addition at analyze press we check connection date

File: CogsMinimizer/Web.config [hunk 10]
```diff
@@ -10,8 +10,6 @@
   </configSections>
   <connectionStrings>
     <add name="DefaultConnection" connectionString="Data Source=(LocalDb)\MSSQLLocalDB;AttachDbFilename=|DataDirectory|\CogsMinimizer.mdf;Initial Catalog=CogsMinimizer_db;Integrated Security=True" providerName="System.Data.SqlClient" />
-      <add name="DataAccess"
```
Comment (shklar) [inline CogsMinimizer/Web.config]: Why did you remove this?
